### PR TITLE
【Hackathon 7th PPSCI No.4】NO.4 DrivAerNet 论文复现

### PR DESCRIPTION
**Pull request by LilaKen**

### CODE REVIEW

File: rfcs/APIs/20241105_api_design_for_DrivAerNet [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+| 文件名                       | 20241105_api_design_for_DrivAerNet.md |  
+
+# 一、概述
+## 1、相关背景
+在汽车设计中，气动性能直接影响到燃油效率、行驶稳定性及安全性等多个方面。过去通常使用基于数值方法的计算流体力学（CFD）或风洞实验来优化设计，但这些方法成本高昂、耗时长。在深度学习迅速发展的背景下，基于数据驱动的方法可以通过学习几何与气动性能之间的关系，快速估算设计的气动性能。
+
+## 2、功能目标
+本次任务的目标是复现基于 DrivAer 数据集的 RegDGCNN 网络模型。DrivAer 数据集包含 4000 个数据样本，设计的网络可以对不同汽车几何结构进行高效的气动阻力预测。通过本项目，将实现以下具体功能：
+1. 复现 DrivAer 数据集上的 RegDGCNN 模型。
+2. 实现与论文一致的精度。
```
Comment (wangguan1995) [inline rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: image
如果可以，最好做一组非全量数据的实验

File: rfcs/APIs/20241105_api_design_for_DrivAerNet [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## 2. 主体设计选型考量
+在 Pytorch 版本中测试了部分 DrivAer 数据，并针对模型复杂度和运行效率进行了调优。飞桨版本的实现考虑了以下选型因素：
+- **模型的复杂度**：采用多层卷积结构，对高维数据进行高效特征提取。
+- **数据量的规模**：针对 DrivAer 的 4000 个样本设计了适配的批处理和数据加载策略。
+- **可扩展性**：考虑到未来其他数据集的适配需求，采用模块化设计方式，便于扩展。
+
+# 五、测试和验收的考量
+在复现的 DrivAerNet 模型上，设置以下测试标准：
+1. **精度验证**：确保飞桨版本与原 Pytorch 版本的相对误差小于 10%，并对特定情况下的误差情况进行分析，确保实际应用的精确性。
```
Comment (wangguan1995) [inline rfcs/APIs/20241105_api_design_for_DrivAerNet:50]: 精度验证主要是R2 score 和 Cd的平均MRE
两个数据分别说明了误差的方差和平均值

**Review by wangguan1995 — COMMENTED** [review]
TO FIX

File: rfcs/APIs/20241105_api_design_for_DrivAerNet (hunk 1 above)
Comment (LilaKen) [reply rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: RegDGCNN复现效果与原论文差一个数量级：

以下是batch_size为14的情况，梯度累积4个batch_size，其他实验参数保留一致。

Average Validation R²: -0.0520
Total training time: 71190.35s
Model saved to models/CdPrediction_DrivAerNet_r2_200epochs_5k_bs14_final_model.pth
Testing the final model:
Test MSE: 0.000737, Test MAE: 0.023203, Max MAE: 0.033418, Test R²: -0.0555
Total inference time: 0.60s for 588 samples
Testing the best model:
Test MSE: 0.000732, Test MAE: 0.023097, Max MAE: 0.033127, Test R²: -0.0482
Total inference time: 0.72s for 588 samples

以下是batch_size为4的情况，epoch为400，其他实验参数保留一致。
Total training time: 261648.30s
Model saved to models/CdPrediction_DrivAerNet_r2_400epochs_5k_final_model.pth
Testing the final model:
Test MSE: 0.000748, Test MAE: 0.023476, Max MAE: 0.040508, Test R²: -1.2606
Total inference time: 1.44s for 592 samples
Testing the best model:
Test MSE: 0.000741, Test MAE: 0.023274, Max MAE: 0.040512, Test R²: -1.2919
Total inference time: 1.54s for 592 samples

Comment (LilaKen) [reply rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: 已同导师沟通完毕，继续复现工作

Comment (Venter-aaa) [reply rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: 我想问一下，从论文的什么部分可以知道我的复现效果怎么样？
这是batch_size为16的训练情况，其他参数保持一致
Model saved to models/cdPrediction DrivAerNet r2 100epochs 5k final model.pth
Testing the final model:
Test MSE: 0.000735, Test MAE: 0.023243, MaX MAE: 0.032163, Test R?: -0.0782
Total inference time: 1.38s for 592 samples
Testing the best model:
Test MSE: 0.000734,TeSt MAE: 0.023104, Max MAE: 0.032490, Test R2: -0.0726
Total inference time: 1.02s for 592 samples

Comment (Venter-aaa) [reply rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: > RegDGCNN复现效果与原论文差一个数量级：
> 
> 以下是batch_size为14的情况，梯度累积4个batch_size，其他实验参数保留一致。
> 
> Average Validation R²: -0.0520 Total training time: 71190.35s Model saved to models/CdPrediction_DrivAerNet_r2_200epochs_5k_bs14_final_model.pth Testing the final model: Test MSE: 0.000737, Test MAE: 0.023203, Max MAE: 0.033418, Test R²: -0.0555 Total inference time: 0.60s for 588 samples Testing the best model: Test MSE: 0.000732, Test MAE: 0.023097, Max MAE: 0.033127, Test R²: -0.0482 Total inference time: 0.72s for 588 samples
> 
> 以下是batch_size为4的情况，epoch为400，其他实验参数保留一致。 Total training time: 261648.30s Model saved to models/CdPrediction_DrivAerNet_r2_400epochs_5k_final_model.pth Testing the final model: Test MSE: 0.000748, Test MAE: 0.023476, Max MAE: 0.040508, Test R²: -1.2606 Total inference time: 1.44s for 592 samples Testing the best model: Test MSE: 0.000741, Test MAE: 0.023274, Max MAE: 0.040512, Test R²: -1.2919 Total inference time: 1.54s for 592 samples

Comment (Venter-aaa) [reply rfcs/APIs/20241105_api_design_for_DrivAerNet:18]: > RegDGCNN复现效果与原论文差一个数量级：
> 
> 以下是batch_size为14的情况，梯度累积4个batch_size，其他实验参数保留一致。
> 
> Average Validation R²: -0.0520 Total training time: 71190.35s Model saved to models/CdPrediction_DrivAerNet_r2_200epochs_5k_bs14_final_model.pth Testing the final model: Test MSE: 0.000737, Test MAE: 0.023203, Max MAE: 0.033418, Test R²: -0.0555 Total inference time: 0.60s for 588 samples Testing the best model: Test MSE: 0.000732, Test MAE: 0.023097, Max MAE: 0.033127, Test R²: -0.0482 Total inference time: 0.72s for 588 samples
> 
> 以下是batch_size为4的情况，epoch为400，其他实验参数保留一致。 Total training time: 261648.30s Model saved to models/CdPrediction_DrivAerNet_r2_400epochs_5k_final_model.pth Testing the final model: Test MSE: 0.000748, Test MAE: 0.023476, Max MAE: 0.040508, Test R²: -1.2606 Total inference time: 1.44s for 592 samples Testing the best model: Test MSE: 0.000741, Test MAE: 0.023274, Max MAE: 0.040512, Test R²: -1.2919 Total inference time: 1.54s for 592 samples

我想问一下，从论文的哪一部分可以看出我的复现效果？
以下是batch_size为16的情况，其他实验参数保持一致。
Model saved to models/cdPrediction DrivAerNet r2 100epochs 5k final model.pth
Testing the final model:
Test MSE: 0.000735, Test MAE: 0.023243, MaX MAE: 0.032163, Test R?: -0.0782
Total inference time: 1.38s for 592 samples
Testing the best model:
Test MSE: 0.000734,TeSt MAE: 0.023104, Max MAE: 0.032490, Test R2: -0.0726
Total inference time: 1.02s for 592 samples

**Pull request merged**